hub/api/tests/test_dataset_handler.py

```diff
@@ -0,0 +1,59 @@
+from hub.util.exceptions import DatasetHandlerError
+from click.testing import CliRunner
+import pytest
+import hub
+import numpy as np
+
+
+def test_new_dataset():
+    with CliRunner().isolated_filesystem():
+        ds = hub.dataset("test_dataset_handler/test_new_dataset")
+        with ds:
+            ds.create_tensor("image")
+            for i in range(10):
+                ds.image.append(i * np.ones((100 * (i + 1), 100 * (i + 1))))
+
+        for i in range(10):
+            np.testing.assert_array_equal(
+                ds.image[i].numpy(), i * np.ones((100 * (i + 1), 100 * (i + 1)))
+            )
```

Comment:
is `hub.dataset` supposed to raise an error when the dataset doesn't exist? can we add a test for this failure case?

Reply:
Nope, hub.dataset creates a new one if it doesn't exist